### PR TITLE
`Micro.EnvRef` and `Micro.Handle` is subtype of `Micro`

### PR DESCRIPTION
**Pull request by KhraksMamtsov**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
`Micro.EnvRef` and `Micro.Handle` is subtype of `Micro`
`MicroableClass` has been exported as an analogue of `Effectable.Class`

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: packages/effect/src/Micro.ts [hunk 1]
```diff
@@ -144,6 +144,45 @@ export interface MicroIterator<T extends Micro<any, any, any>> {
...
+  Base.prototype = MicroProto
+  return Base as any
+})()
+
+/**
+ * @since 3.9.0
+ * @category constructors
+ */
+export abstract class MicroableClass<out A, out E = never, out R = never> extends MicroBase<A, E, R> {
+  abstract readonly [runSymbol]: (env: Env<any>, onExit: (exit: MicroExit<A, E>) => void) => void
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:183]:
```suggestion
  abstract [runSymbol](env: Env<any>, onExit: (exit: MicroExit<A, E>) => void): void
```

File: packages/effect/src/Micro.ts [hunk 2]
```diff
@@ -144,6 +144,45 @@ export interface MicroIterator<T extends Micro<any, any, any>> {
...
+  function Base() {}
+  Base.prototype = MicroProto
+  return Base as any
+})()
+
+/**
+ * @since 3.9.0
+ * @category constructors
+ */
+export abstract class MicroableClass<out A, out E = never, out R = never> extends MicroBase<A, E, R> {
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:182]: I think `Class` is fine. `MicroableClass` is a bit of a mouthful,  and the usage could be:

```ts
import { Micro } from "effect"

class Thing extends Micro.Class<A, E> {
  // ...
}
```

File: packages/effect/src/Micro.ts [hunk 3]
```diff
@@ -3671,7 +3741,14 @@ export interface Handle<A, E = never> {
 export const isHandle = (u: unknown): u is Handle<unknown, unknown> =>
   typeof u === "object" && u !== null && HandleTypeId in u
 
-class HandleImpl<A, E> implements Handle<A, E> {
+class HandleImpl<A, E> extends MicroableClass<A, E> implements Handle<A, E> {
+  readonly [runSymbol] = (_env: Env<any>, onExit: (exit: MicroExit<A, E>) => void) => {
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:3745]:
```suggestion
  [runSymbol](_env: Env<any>, onExit: (exit: MicroExit<A, E>) => void) {
```

File: packages/effect/src/Micro.ts [hunk 4]
```diff
@@ -144,6 +144,45 @@ export interface MicroIterator<T extends Micro<any, any, any>> {
   next(...args: ReadonlyArray<any>): IteratorResult<YieldWrap<T>, Micro.Success<T>>
 }
 
+/**
+ * @since 3.9.0
+ * @category models
+ */
+export interface MicroPrimitive {
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:151]: I think we can call this `MicroClass`.

File: packages/effect/src/Micro.ts [hunk 5]
```diff
@@ -3671,7 +3741,14 @@ export interface Handle<A, E = never> {
 export const isHandle = (u: unknown): u is Handle<unknown, unknown> =>
   typeof u === "object" && u !== null && HandleTypeId in u
 
-class HandleImpl<A, E> implements Handle<A, E> {
+class HandleImpl<A, E> extends MicroableClass<A, E> implements Handle<A, E> {
+  readonly [runSymbol] = (_env: Env<any>, onExit: (exit: MicroExit<A, E>) => void) => {
+    const observer = (exit: MicroExit<A, E>) => {
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:3746]: This should handle interruption too.

File: packages/effect/src/Micro.ts (hunk 1 above)
Comment (KhraksMamtsov) [reply packages/effect/src/Micro.ts]: done

File: packages/effect/src/Micro.ts (hunk 3 above)
Comment (KhraksMamtsov) [reply packages/effect/src/Micro.ts]: done

File: packages/effect/src/Micro.ts (hunk 4 above)
Comment (KhraksMamtsov) [reply packages/effect/src/Micro.ts]: done

File: packages/effect/src/Micro.ts (hunk 5 above)
Comment (KhraksMamtsov) [reply packages/effect/src/Micro.ts:3750]: Done, but please check it again)

Comment (tim-smart) [reply packages/effect/src/Micro.ts]: I added an abstract `asMicro` method to the Class, as implementing the `runSymbol` function has a bunch of footguns that you have to be aware of.

**Comment by KhraksMamtsov** [comment]
looks like flaky

**Pull request merged**